### ISSUE TITLE
[bug]Specific event bug

### ISSUE THREAD

**Issue by marlangcom** [MaaAssistantArknights/MaaAssistantArknights#6835]
### Before raising the issue...

- [X] I understand that Issues are for feedback and problem solving, not trolling the comments section, and will provide as much information as possible to help solve the problem.
- [X] I filled in a short, clear title so that developers could quickly identify the general problem when going through the issue list. Instead of "Some suggestions", "Stuck", etc.
- [X] I have checked the latest beta update and there is no mention of the bug being fixed.

### Description

Client : YoStarKR
Emu : PlayGames BETA + ADB
MAA VER : MAA-v4.24.0-win-x64

If fail a mission after entering a roguelike and get "SIGNAL LOST", 
will not be able to progress to the next one.

### Log files

[asst.bak_2.log](https://github.com/MaaAssistantArknights/MaaAssistantArknights/files/12850678/asst.bak_2.log)

### Screenshots

![image](https://github.com/MaaAssistantArknights/MaaAssistantArknights/assets/72522917/cd382eaf-ad01-4c3a-bd9e-b9c46f4991e4)
![image](https://github.com/MaaAssistantArknights/MaaAssistantArknights/assets/72522917/08ad1e84-7664-489c-a61a-50b230c4a40c)

**Comment by zzyyyl** [comment]
MAA has tried to match `MissionFailedFlag` 20 times and failed. 
`MissionFailedFlag` is: ![](https://github.com/MaaAssistantArknights/MaaAssistantArknights/blob/dev/resource/template/Phantom@Roguelike@MissionFailedFlag.png?raw=true)

**Comment by zzyyyl** [comment]
Maybe it's something like #6584 ?

**Comment by zzyyyl** [comment]
<details>
<summary>main log</summary>

```
[2023-10-10 05:41:22.332][TRC][Px5674][Tx536c] asst::ProcessTask::_run | enter
[2023-10-10 05:41:22.332][INF][Px5674][Tx536c] {"class":"asst::ProcessTask","details":{"cur_retry":20,"retry_times":20,"to_be_recognized":["Phantom@Roguelike@InBattleFlag","Phantom@Roguelike@MissionCompletedFlag","Phantom@Roguelike@MissionFailedFlag","Phantom@Roguelike@MissionFailedFlag2","Phantom@Roguelike@GamePassSkip1","Phantom@Roguelike@GamePassTheEndConfirm","Phantom@Roguelike@GamePass","Phantom@Roguelike@StartAction@LoadingText","Phantom@Roguelike@StartAction@LoadingIcon"]},"first":["Phantom@Roguelike@Begin"],"pre_task":"Phantom@Roguelike@StartAction","subtask":"ProcessTask","taskchain":"Roguelike","taskid":4}
[2023-10-10 05:41:22.463][INF][Px5674][Tx536c] Call ` C:\명빵\PlayBridge.exe -s 127.0.0.1:5000 exec-out screencap -p ` ret 0 , cost 130 ms , stdout size: 722850 , socket size: 0
[2023-10-10 05:41:22.481][TRC][Px5674][Tx536c] asst::PipelineAnalyzer::analyze Phantom@Roguelike@InBattleFlag
[2023-10-10 05:41:22.482][TRC][Px5674][Tx536c] asst::PipelineAnalyzer::analyze Phantom@Roguelike@MissionCompletedFlag
[2023-10-10 05:41:22.482][TRC][Px5674][Tx536c] asst::PipelineAnalyzer::analyze Phantom@Roguelike@MissionFailedFlag
[2023-10-10 05:41:22.491][TRC][Px5674][Tx536c] asst::PipelineAnalyzer::analyze Phantom@Roguelike@MissionFailedFlag2
[2023-10-10 05:41:22.495][TRC][Px5674][Tx536c] asst::PipelineAnalyzer::analyze Phantom@Roguelike@GamePassSkip1
[2023-10-10 05:41:22.496][TRC][Px5674][Tx536c] asst::PipelineAnalyzer::analyze Phantom@Roguelike@GamePassTheEndConfirm
[2023-10-10 05:41:22.498][TRC][Px5674][Tx536c] match_templ | Roguelike@GamePassTheEndConfirm.png score: 0.528215 rect: [ 603, 588, 77, 61 ] roi: [ 554, 512, 177, 161 ]
[2023-10-10 05:41:22.498][TRC][Px5674][Tx536c] asst::PipelineAnalyzer::analyze Phantom@Roguelike@GamePass
[2023-10-10 05:41:22.509][TRC][Px5674][Tx536c] asst::PipelineAnalyzer::analyze Phantom@Roguelike@StartAction@LoadingText
[2023-10-10 05:41:22.528][TRC][Px5674][Tx536c] asst::WordOcr [{ _: [ 0, 0, 300, 120 ], score: 0.757101 }] by OCR Pipeline , cost 18 ms
[2023-10-10 05:41:22.528][TRC][Px5674][Tx536c] Proced []
[2023-10-10 05:41:22.528][TRC][Px5674][Tx536c] asst::PipelineAnalyzer::analyze Phantom@Roguelike@StartAction@LoadingIcon
[2023-10-10 05:41:22.535][TRC][Px5674][Tx536c] asst::ProcessTask::_run | leave, 203 ms
[2023-10-10 05:41:22.535][TRC][Px5674][Tx536c] ready to sleep 500
[2023-10-10 05:41:23.044][TRC][Px5674][Tx536c] end of sleep 500
[2023-10-10 05:41:23.044][INF][Px5674][Tx536c] Assistant::append_callback | SubTaskStart {"class":"asst::RoguelikeDebugTaskPlugin","details":{},"subtask":"RoguelikeDebugTask","taskchain":"Roguelike","taskid":4,"uuid":""}
[2023-10-10 05:41:23.158][INF][Px5674][Tx536c] Call ` C:\명빵\PlayBridge.exe -s 127.0.0.1:5000 exec-out screencap -p ` ret 0 , cost 114 ms , stdout size: 722850 , socket size: 0
[2023-10-10 05:41:23.176][TRC][Px5674][Tx536c] Save image debug\roguelike\2023-10-10_05-41-23-176_raw.png
[2023-10-10 05:41:23.204][INF][Px5674][Tx536c] Assistant::append_callback | SubTaskCompleted 
```
</details>

**Comment by zzyyyl** [comment]
@marlangcom Check if your `resource\template\Phantom@Roguelike@MissionFailedFlag.png` looks similar to https://github.com/MaaAssistantArknights/MaaAssistantArknights/issues/6835#issuecomment-1754442285 and make sure there is no `resource\global\YoStarKR\resource\template\Phantom@ Roguelike@MissionFailedFlag.png`.

**Comment by Constrat** [comment]
> Maybe it's something like #6584 ?

Let's hope not... Do you know if I can build a Non debug version of MAA with VS?
Because I want to try a non-debug version, maybe it's faster => the issue happens more

**Comment by sodorisa** [comment]
[2023-10-10 17:46:24.703][INF][Pxcd8][Txca0] {"class":"asst::ProcessTask","details":{"cur_retry":0,"retry_times":20,"to_be_recognized":["Phantom@Roguelike@InBattleFlag","Phantom@Roguelike@MissionCompletedFlag","Phantom@Roguelike@MissionFailedFlag","Phantom@Roguelike@MissionFailedFlag2","Phantom@Roguelike@GamePassSkip1","Phantom@Roguelike@GamePassTheEndConfirm","Phantom@Roguelike@GamePass","Phantom@Roguelike@StartAction@LoadingText","Phantom@Roguelike@StartAction@LoadingIcon"]},"first":["Phantom@Roguelike@Begin"],"pre_task":"Phantom@Roguelike@StartAction","subtask":"ProcessTask","taskchain":"Roguelike","taskid":2}
[2023-10-10 17:46:24.903][INF][Pxcd8][Txca0] Call ` C:\Program Files\BlueStacks_nxt\.\HD-Adb.exe -s 127.0.0.1:1409 exec-out "screencap | gzip -1" ` ret 0 , cost 200 ms , stdout size: 1778326 , socket size: 0
[2023-10-10 17:46:24.957][TRC][Pxcd8][Txca0] asst::PipelineAnalyzer::analyze Phantom@Roguelike@InBattleFlag
[2023-10-10 17:46:24.964][TRC][Pxcd8][Txca0] asst::PipelineAnalyzer::analyze Phantom@Roguelike@MissionCompletedFlag
[2023-10-10 17:46:24.966][TRC][Pxcd8][Txca0] asst::PipelineAnalyzer::analyze Phantom@Roguelike@MissionFailedFlag
[2023-10-10 17:46:25.029][TRC][Pxcd8][Txca0] asst::PipelineAnalyzer::analyze Phantom@Roguelike@MissionFailedFlag2
[2023-10-10 17:46:25.055][TRC][Pxcd8][Txca0] asst::PipelineAnalyzer::analyze Phantom@Roguelike@GamePassSkip1
[2023-10-10 17:46:25.068][TRC][Pxcd8][Txca0] asst::PipelineAnalyzer::analyze Phantom@Roguelike@GamePassTheEndConfirm
[2023-10-10 17:46:25.084][TRC][Pxcd8][Txca0] match_templ | Roguelike@GamePassTheEndConfirm.png score: 0.531397 rect: [ 602, 588, 77, 61 ] roi: [ 554, 512, 177, 161 ]
[2023-10-10 17:46:25.084][TRC][Pxcd8][Txca0] asst::PipelineAnalyzer::analyze Phantom@Roguelike@GamePass
[2023-10-10 17:46:25.134][TRC][Pxcd8][Txca0] asst::PipelineAnalyzer::analyze Phantom@Roguelike@StartAction@LoadingText

It seems to recognize a white check mark rather than a signal lost.
There is no check mark on Mizuki Roguelike's battle failure screen, and it works fine.

**Comment by marlangcom** [comment]
> @marlangcom Check if your `resource\template\Phantom@Roguelike@MissionFailedFlag.png` looks similar to [#6835 (comment)](https://github.com/MaaAssistantArknights/MaaAssistantArknights/issues/6835#issuecomment-1754442285) and make sure there is no `resource\global\YoStarKR\resource\template\Phantom@ Roguelike@MissionFailedFlag.png`.

I tried again with MAA version 4.25.0.
I played the original version without the .png file added.
[asst.bak_addimageMAA4.25.0.log](https://github.com/MaaAssistantArknights/MaaAssistantArknights/files/12856672/asst.bak_addimageMAA4.25.0.log)
and the other with the .png file added.
[asst.bak_originalMAA4.25.0.log](https://github.com/MaaAssistantArknights/MaaAssistantArknights/files/12856673/asst.bak_originalMAA4.25.0.log)

The results are the same, MAA stops progressing when the game ends in the middle of a phantom roguelike.

I think it has something to do with the checkmarks not appearing in the Mizuki roguelike, but I'm not sure.
